src/main/java/io/opentracing/contrib/global/GlobalSpanManager.java

```diff
@@ -0,0 +1,87 @@
+package io.opentracing.contrib.global;
+
+import io.opentracing.Span;
+
+import java.io.Closeable;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This class manages the globally 'active' {@link Span} in the current Thread.
+ *
+ * @author Sjoerd Talsma
+ */
+public final class GlobalSpanManager {
+
+    private static final Logger LOGGER = Logger.getLogger(GlobalSpanManager.class.getName());
+
+    private static final ThreadLocal<ManagedSpan> ACTIVE_SPAN = new InheritableThreadLocal<ManagedSpan>();
+
+    /**
+     * Static method to return the currently active global {@link Span}.<br>
+     * There is no guarantee that there is an active {@link Span} in all situations.
+     *
+     * @return The currently active global Span, or <code>null</code> if there is no Span currently active.
+     */
+    public static Span activeSpan() {
+        final ManagedSpan active = ACTIVE_SPAN.get();
+        return active != null ? active.span : null; // TODO: Or return no-op Span instead?
+    }
+
+    public static GlobalSpanDeactivator activate(Span span) {
```

Comment:
Another approach would be to take a `Runnable` or similar as a second parameter and make the activation/deactivation private to this class. That would be a little less flexible and a little more verbose, but it would also be impossible to forget to "deactivate".

Yet another option would be to create something like a `TracedRunnable` version of `Runnable`: that is essentially what happened for Dapper in Java. Many of the threadlocal pieces were similar.